### PR TITLE
Add controller for invitation e-mail sending

### PR DESCRIPTION
**Pull request by HappyTetrahedron**
## Summary
Adds a controller for e-mail sending. By default, it logs to stdout instead of sending an e-mail. 

Using command line parameters, it can be configured to send actual emails via Mailgun instead, optionally using Mailgun's test mode.

Emails are only sent for invitations which have the EmailSent condition explicitly set to `false`. Email sending errors are added to the condition, and the sending is retried at a configurable interval.

Implements APUB-398

Documentation: https://wiki.vshn.net/pages/viewpage.action?pageId=328567805

...

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [x] Link this PR to related issues or PRs.

### CODE REVIEW

File: controller.go [hunk 1]
```diff
@@ -58,6 +59,17 @@ func ControllerCommand() *cobra.Command {
...
+	invEmailBackend := cmd.Flags().String("email-backend", "stdout", "Backend to use for sending invitation mails (one of stdout, mailgun)")
+	invEmailSender := cmd.Flags().String("email-sender", "noreply@appuio.cloud", "Sender address for invitation mails")
+	invEmailSubject := cmd.Flags().String("email-subject", "You have been invited to APPUiO Cloud", "Subject for invitation mails")
+	invEmailRetryInterval := cmd.Flags().Duration("email-retry-interval", 5*time.Minute, "Retry interval for sending e-mail messages")
+
+	invEmailMailgunToken := cmd.Flags().String("mailgun-token", "CHANGEME", "Token used to access Mailgun API")
+	invEmailMailgunDomain := cmd.Flags().String("mailgun-domain", "example.com", "Mailgun Domain to use")
+	invEmailMailgunTemplate := cmd.Flags().String("mailgun-template", "appuio-cloud-invitation", "Name of the Mailgun template")
+	invEmailMailgunUrl := cmd.Flags().String("mailgun-url", "https://api.eu.mailgun.net/v3", "API base URL for your Mailgun account")
+	invEmailMailgunDebug := cmd.Flags().Bool("mailgun-debug", false, "If set, do not actually send e-mails")
```
Comment (bastjan) [inline controller.go:71]: For me `debug` sounds more like show more info. I'd prefer test mode or `dry-run` or don't send.

File: controllers/invitation_email_controller.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+			Type:    userv1.ConditionEmailSent,
+			Status:  metav1.ConditionTrue,
+			Message: fmt.Sprintf("Message ID: %s", id),
+		})
+	} else {
+		apimeta.SetStatusCondition(&inv.Status.Conditions, metav1.Condition{
+			Type:   userv1.ConditionEmailSent,
+			Status: metav1.ConditionTrue,
+		})
+	}
```
Comment (bastjan) [inline controllers/invitation_email_controller.go:93]: Might be a bit shorter.
```suggestion
	var message string
	if id != "" {
		message = fmt.Sprintf("Message ID: %s", id)
	}
	apimeta.SetStatusCondition(&inv.Status.Conditions, metav1.Condition{
		Type:       userv1.ConditionEmailSent,
		Status:     metav1.ConditionTrue,
		Message: message,
	})
```

File: go.mod [hunk 3]
```diff
@@ -24,6 +24,8 @@ require (
 	sigs.k8s.io/kustomize/kustomize/v4 v4.5.7
 )
 
+require github.com/gorilla/mux v1.8.0 // indirect
```
Comment (bastjan) [inline go.mod:27]: This might need a `go mod tidy`.

**Review by bastjan — APPROVED** [review]
Looks very good

File: controller.go (hunk 1 above)
Comment (HappyTetrahedron) [reply controller.go:71]: Very fair point

**Pull request merged**